core/views.py

```diff
@@ -47,6 +48,54 @@ def get_queryset(self: "SavedPostListView") -> models.QuerySet:
         )
 
 
+class UserPostListView(LoginRequiredMixin, ListView):
+    template_name = "core/user-post-list.html"
+    context_object_name = "posts"
+
+    def get_queryset(self: "UserPostListView") -> models.QuerySet:
+        user_post = Post.objects.filter(author=self.request.user).order_by("-created_at")
+        status = self.request.GET.get("status")
+        if status == "draft":
```

Comment:
Please use match case: 
https://akademiait.com.pl/jak-zaimplentowac-switch-case-w-pythonie/ section: Python 3.10 i dopasowywanie wzorców strukturalnych.

Reply:
Or you could use a dict, which is bit more complicated but would be much shorter syntax.